### ISSUE TITLE
Output capture decorator

### ISSUE THREAD

**Issue by jasongrout** [jupyter-widgets/ipywidgets#1846]
It's a little awkward making a callback function who's output should be captured:

```python
from ipywidgets import Output, IntSlider, VBox
from IPython.display import display
out = Output()
slider = IntSlider()
def change(change):
    with out:
        print('value: '+change.new)
slider.observe(change, 'value')
display(VBox([slider, out]))
```

Some people probably would prefer to have a decorator that automatically captures output:

```python
from ipywidgets import Output, IntSlider, VBox
from IPython.display import display
out = Output()
slider = IntSlider()

@out.capture() # new decorator - not sure if this is a good name for it...
def change(change):
    print('value: '+change.new)

slider.observe(change, 'value')
display(VBox([slider, out]))
```

**Comment by jasongrout** [comment]
(this would be similar to https://docs.python.org/3/library/contextlib.html#using-a-context-manager-as-a-function-decorator)

**Comment by jasongrout** [comment]
(Note that I don't think we should necessarily use the ContextDecorator, just that that is one way to implement the idea)

**Comment by hainm** [comment]
```python
@out.capture()
def change(change):
    print('value: '+change.new)
```

nice

**Comment by jasongrout** [comment]
Fixed by #1934. Thanks @pbugnion!
